metricbeat/module/system/filesystem/helper.go

```diff
@@ -31,19 +31,45 @@ func GetFileSystemList() ([]sigar.FileSystem, error) {
 		return nil, err
 	}
 
-	// Ignore relative mount points, which are present for example
-	// in /proc/mounts on Linux with network namespaces.
-	filtered := fss.List[:0]
-	for _, fs := range fss.List {
-		if filepath.IsAbs(fs.DirName) {
+	return filterFileSystemList(fss.List), nil
+}
+
+// filterFileSystemList filters mountpoints to avoid virtual filesystems
+// and duplications
+func filterFileSystemList(fsList []sigar.FileSystem) []sigar.FileSystem {
+	var filtered []sigar.FileSystem
+	devices := make(map[string]sigar.FileSystem)
+	for _, fs := range fsList {
+		// Ignore relative mount points, which are present for example
+		// in /proc/mounts on Linux with network namespaces.
+		if !filepath.IsAbs(fs.DirName) {
+			debugf("Filtering filesystem with relative mountpoint %+v", fs)
+			continue
+		}
+
+		// Don't do further checks in special devices
+		if !filepath.IsAbs(fs.DevName) {
 			filtered = append(filtered, fs)
 			continue
 		}
-		debugf("Filtering filesystem with relative mountpoint %+v", fs)
+
+		// If a block device is mounted multiple times (e.g. with bind mounts),
+		// store it only once, and use the shorter mount point path.
+		if seen, found := devices[fs.DevName]; found {
```

Comment:
This is not completely accurate, there can be different device paths that are really pointing to the same block device, but I don't believe this is common at all. To control this we should check duplicates by device identifier (`Major:Minor`) but this is not supported in gosigar yet. Does it worth to implement this?

Reply:
I would leave it what you have for now but potentially follow up with a change in gosigar to make it possible if it is a case that can happen (more then once :-)). @andrewkroh WDYT?

Reply:
If it will help make Metricbeat more accurate then I'm +1.

Reply:
I will take a look to this in gosigar, meanwhile I think we are good enough with the checks based on device name.